### ISSUE TITLE
Errata in examples 2.1 and 2.4

### ISSUE THREAD

**Issue by Joaoloula** [nilearn/nilearn#1130]
In section [2.1.3.1,](http://nilearn.github.io/decoding/decoding_tutorial.html#dimension-reduction-with-feature-selection) the pipeline model is used to fit and predict _X_ and _y_ variables that weren't declared before in the tutorial: it seems these were taken from the example _plot_haxby_anova_svm.py_, so something like this should probably be added:

`mask_filename = haxby_dataset.mask`
`nifti_masker = NiftiMasker(mask_img=mask_filename, standardize=True)`
`func_filename = haxby_dataset.func[0]`
`X = nifti_masker.fit_transform(func_filename)`
`X = X[condition_mask]`

with eventually a change in the name of the mask variables just to make it clear that they're different from those used at the beginning of the section. The _y_ variable can just be replaced with the _target_ variable declared before.

In the first paragraph of section [2.4.1](http://nilearn.github.io/decoding/space_net.html#the-spacenet-decoder), both links to [Grosenick et al. 2013] redirect to [Gramfort et al. 2013], they should probably be changed to http://www.ncbi.nlm.nih.gov/pubmed/23298747.

**Comment by KamalakerDadi** [comment]
>the pipeline model is used to fit and predict X and y variables that weren't declared before in the tutorial:

The idea of this section being that more focussed on how to do feature selection procedure providing an idea. For me, I think user will know the meaning of X and y in fit and predict (as it explained above).

>In the first paragraph of section 2.4.1, both links to [Grosenick et al. 2013] redirect to [Gramfort et al. 2013], they should probably be changed to http://www.ncbi.nlm.nih.gov/pubmed/23298747.

Thanks for spotting. PR welcome. :)

**Comment by Joaoloula** [comment]
Ok, I'll do a PR

**Comment by dohmatob** [comment]
On Tue, Jun 14, 2016 at 11:05 PM, Joao Loula <notifications@github.com>
wrote:

> In section 2.1.3.1,
> <http://nilearn.github.io/decoding/decoding_tutorial.html#dimension-reduction-with-feature-selection>
> the pipeline model is used to fit and predict *X* and *y* variables that
> weren't declared before in the tutorial: it seems these were taken from the
> example *plot_haxby_anova_svm.py*, so something like this should probably
> be added:
>
> mask_filename = haxby_dataset.mask
> nifti_masker = NiftiMasker(mask_img=mask_filename, standardize=True)
> func_filename = haxby_dataset.func[0]
> X = nifti_masker.fit_transform(func_filename)
> X = X[condition_mask]
>
> with eventually a change in the name of the mask variables just to make it
> clear that they're different from those used at the beginning of the
> section. The *y* variable can just be replaced with the *target* variable
> declared before.
>
> In the first paragraph of section 2.4.1
> <http://nilearn.github.io/decoding/space_net.html#the-spacenet-decoder>,
> both links to [Grosenick et al. 2013] redirect to [Gramfort et al. 2013],
> they should probably be changed to
> http://www.ncbi.nlm.nih.gov/pubmed/23298747.
>
Good catch.

> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nilearn/nilearn/issues/1130>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAms1Ia4nL3lNUBaf2xnNnP68LD2PxESks5qLxevgaJpZM4I1wqY>
> .
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob

**Comment by tsalo** [comment]
It looks like the link issues were resolved in #1184, so now the only problem is that Example 2.1 (An introduction to decoding) does not define the variables used in section 2.1.3.1 (Dimension reduction with feature selection)- specifically `func_img` and `conditions`. 

The example uses the following lines from `plot_haxby_anova_svm.py`, which does not include the code used to define `func_img` and `conditions`: https://github.com/nilearn/nilearn/blob/0c6388a26950beaeec0925e67b9bfb55ccfa2fde/examples/02_decoding/plot_haxby_anova_svm.py#L56-L86
